File: MyMinimalApi/Account/Hub.cs
using Microsoft.AspNetCore.SignalR;

namespace Bank.Hubs;

public record StateTransitionMessage(object Event, object NewState);

public interface IAccountClient {
   Task ReceiveMessage(StateTransitionMessage stateTransition);
   Task ReceiveError(string error);
}

public class AccountHub : Hub<IAccountClient> {
   public async Task RemoveFromConnectionGroup(string accountId) =>
      await Groups.RemoveFromGroupAsync(Context.ConnectionId, accountId);

   public async Task AddToConnectionGroup(string accountId) =>
      await Groups.AddToGroupAsync(Context.ConnectionId, accountId);
}

File: Archive/FSharpWithAkka/Account/Hub.fs
namespace Bank.Hubs

open Microsoft.AspNetCore.SignalR
open System.Threading.Tasks

open BankTypes

type IAccountClient =
   abstract member ReceiveMessage:
      {|
         event: AccountEvent
         newState: AccountState
      |} ->
         Task

   abstract member ReceiveError: string -> Task

   abstract member ReceiveCircuitBreakerMessage: CircuitBreakerMessage -> Task

type AccountHub() =
   inherit Hub<IAccountClient>()

   member x.RemoveFromConnectionGroup(accountId: string) =
      x.Groups.RemoveFromGroupAsync(x.Context.ConnectionId, accountId)

   member x.AddToConnectionGroup(accountId: string) =
      x.Groups.AddToGroupAsync(x.Context.ConnectionId, accountId)
